pkg/driver/trace_driver.go

```diff
@@ -665,34 +597,14 @@ func (d *Driver) RunExperiment(skipIATGeneration bool, readIATFromFIle bool) {
 
 	trace.ApplyResourceLimits(d.Configuration.Functions, d.Configuration.LoaderConfiguration.CPULimit)
 
-	switch d.Configuration.LoaderConfiguration.Platform {
-	case "Knative", "Knative-RPS":
-		DeployFunctions(d.Configuration.Functions,
-			d.Configuration.YAMLPath,
-			d.Configuration.LoaderConfiguration.IsPartiallyPanic,
-			d.Configuration.LoaderConfiguration.EndpointPort,
-			d.Configuration.LoaderConfiguration.AutoscalingMetric)
-		go scheduleFailure(d.Configuration.LoaderConfiguration)
-	case "OpenWhisk", "OpenWhisk-RPS":
-		DeployFunctionsOpenWhisk(d.Configuration.Functions)
-	case "AWSLambda", "AWSLambda-RPS":
-		DeployFunctionsAWSLambda(d.Configuration.Functions)
-	case "Dirigent", "Dirigent-RPS", "Dirigent-Dandelion", "Dirigent-Dandelion-RPS":
-		DeployDirigent(d.Configuration.LoaderConfiguration.DirigentControlPlaneIP, d.Configuration.Functions)
-		go scheduleFailure(d.Configuration.LoaderConfiguration)
-	default:
-		log.Fatal("Unsupported platform.")
-	}
+	deployer, configuration := deployment.CreateDeployer(d.Configuration)
```

Comment:
You might have to have one single returned object. Configuration is only used in the scope of the deployer. Therefore you can create it locally only.